mobilecoind/strategies/drain-accounts.py

```diff
@@ -85,20 +85,55 @@ def run_test(stub, amount, monitor_id, dest, max_seconds, token_id):
         mobilecoind_api_pb2.GetBalanceRequest(monitor_id=monitor_id, token_id=token_id))
     starting_balance = resp.balance
     logging.info("Starting balance prior to transfer: %s", starting_balance)
-    tx_resp = stub.SendPayment(
-        mobilecoind_api_pb2.SendPaymentRequest(
-            sender_monitor_id=monitor_id,
-            sender_subaddress=0,
-            outlay_list=[
-                mobilecoind_api_pb2.Outlay(
-                    value=amount,
-                    receiver=dest,
-                )
-            ],
-            fee=0,
-            tombstone=0,
-            token_id=token_id,
-        ))
+
+    # Try building a payment that sends all of our balance
+    # We may get a defragmentation error -- if we do, then submit optimization Tx outs
+    # until we don't get that error anymore
+    while True:
+        try:
+            tx_resp = stub.SendPayment(
+                mobilecoind_api_pb2.SendPaymentRequest(
+                    sender_monitor_id=monitor_id,
+                    sender_subaddress=0,
+                    outlay_list=[
+                        mobilecoind_api_pb2.Outlay(
+                            value=amount,
+                            receiver=dest,
+                        )
+                    ],
+                    fee=0,
+                    tombstone=0,
+                    token_id=token_id,
+                ))
+            break
+         except grpc.RpcError as e:
+            if "Insufficient funds due to UTXO fragmentation" in e.details:
```

Comment:
This perhaps too specific, this strategy could break pretty easily.  We should at least do .lower().

```suggestion
            if "insufficient funds due to utxo fragmentation" in e.details.lower():
```